<a href="https://colab.research.google.com/github/Lukas-Howlett/DenverCoffeeShop/blob/main/DenverCoffeeShop.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# A new coffee shop wants to open stores in Denver and believes that the ideal location is near affluent households and that their store appeals most to those aged 20-35.

## We want to find neighborhoods with high populations of people aged 20-35 near affluent neighborhoods.

### Don't have to worry about competition from Starbucks since the coffee shop doesn't particularly compete with them.

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

In [ ]:
coffee_df = pd.read_csv("/content/sample_data/denverCoffeeShop.csv")

In [ ]:
display(coffee_df)

NBHD_ID      NBHD_NAME  POPULATION_2010  AGE_LESS_18  AGE_18_TO_34  \
0         1    Athmar Park           8898.0       2649.0        2406.0   
1         2        Auraria            705.0          7.0         635.0   
2         3          Baker           4879.0        745.0        1762.0   
3         4         Barnum           6111.0       2015.0        1647.0   
4         5    Barnum West           5376.0       1579.0        1380.0   
..      ...            ...              ...          ...           ...   
73       74    West Colfax           9740.0       2756.0        2935.0   
74       75  West Highland           8540.0       1304.0        2555.0   
75       76       Westwood          15486.0       5678.0        4037.0   
76       77       Whittier           4831.0       1119.0        1586.0   
77       78        Windsor          12589.0       1819.0        3302.0   

    AGE_35_TO_65  AGE_65_PLUS  NUM_HOUSEHOLDS  FAMILIES  NUM_HHLD_100K+  
0         3112.0        731.0          2781.0    1964.0           425.0  
1           57.0          6.0            60.0      15.0             NaN  
2         2037.0        335.0          2447.0     833.0             NaN  
3         2089.0        360.0          1782.0    1289.0           120.0  
4         1824.0        593.0          1704.0    1207.0             NaN  
..           ...          ...             ...       ...             ...  
73        3111.0        938.0          3703.0    1971.0           431.0  
74        3756.0        925.0          4256.0    1853.0          1725.0  
75        4712.0       1059.0          4340.0    3306.0           326.0  
76        1788.0        338.0          1993.0    1008.0             NaN  
77        4085.0       3383.0          7204.0    2590.0           723.0  

[78 rows x 10 columns]

In [ ]:
coffee_df[coffee_df['NUM_HHLD_100K+'] > 0].sort_values(by=['AGE_18_TO_34'], ascending=False)

NBHD_ID                     NBHD_NAME  POPULATION_2010  AGE_LESS_18  \
8         9                  Capitol Hill          14708.0        408.0   
27       28  Gateway - Green Valley Ranch          29201.0      10074.0   
44       45                     Montbello          30348.0      11137.0   
25       26                   Five Points          12712.0       1446.0   
63       64                    University           9375.0        826.0   
58       59                         Speer          10954.0        707.0   
31       32                       Hampden          17547.0       2754.0   
75       76                      Westwood          15486.0       5678.0   
59       60                  Central Park          13948.0       3516.0   
71       72      Washington Virginia Vale          13030.0       2252.0   
68       69              Virginia Village          12844.0       2421.0   
32       33                 Hampden South          14370.0       2639.0   
19       20                 Congress Park          10235.0       1427.0   
23       24                   East Colfax          10191.0       2595.0   
77       78                       Windsor          12589.0       1819.0   
35       36                      Highland           8429.0       1296.0   
11       12                 Cheesman Park           7971.0        510.0   
42       43                       Mar Lee          12452.0       3685.0   
46       47            North Capitol Hill           5823.0        194.0   
43       44                       Marston          11132.0       2266.0   
33       34                   Harvey Park          11525.0       3162.0   
65       66               University Park           7491.0       1135.0   
73       74                   West Colfax           9740.0       2756.0   
61       62                     Sunnyside           9726.0       2363.0   
74       75                 West Highland           8540.0       1304.0   
40       41                  Lincoln Park           6119.0       1365.0   
67       68                    Villa Park           8758.0       2741.0   
0         1                   Athmar Park           8898.0       2649.0   
53       54                     Ruby Hill           9820.0       2865.0   
5         6                   Bear Valley           8889.0       2036.0   
70       71          Washington Park West           6393.0        807.0   
7         8                      Berkeley           8112.0       1305.0   
41       42                   Lowry Field           8067.0       2000.0   
34       35             Harvey Park South           8393.0       2065.0   
30       31                          Hale           6936.0        987.0   
47       48               North Park Hill           9382.0       2179.0   
48       49           Northeast Park Hill           7822.0       2198.0   
55       56                    Sloan Lake           7238.0       1324.0   
56       57               South Park Hill           8590.0       1851.0   
26       27                    Fort Logan           8532.0       1796.0   
29       30                     Goldsmith           5808.0       1060.0   
18       19   College View - South Platte           6498.0       2037.0   
24       25                Elyria Swansea           6401.0       2322.0   
3         4                        Barnum           6111.0       2015.0   
69       70               Washington Park           6905.0       1199.0   
64       65              University Hills           5327.0        925.0   
12       13                  Cherry Creek           5589.0        413.0   
36       37                       Hilltop           8190.0       2016.0   

    AGE_18_TO_34  AGE_35_TO_65  AGE_65_PLUS  NUM_HOUSEHOLDS  FAMILIES  \
8         8274.0        5155.0        871.0         10856.0    1201.0   
27        7778.0       10405.0        944.0          9427.0    7056.0   
44        7742.0        9561.0       1908.0          7942.0    6399.0   
25        5961.0        4660.0      

# The chart above displays neighborhoods in Denver with the greatest target population for the Coffee Shop in descending order.

## The top three neighborhoods are **Capitol Hill** with a target population of 8,274, **Gateway - Green Valley Ranch** with a target population of 7,778, and **Montbello** with a target population of 7,742.

In [ ]:
coffee_df[coffee_df['NUM_HHLD_100K+'] > 0].sort_values(by=['NUM_HHLD_100K+'], ascending=False)

NBHD_ID                     NBHD_NAME  POPULATION_2010  AGE_LESS_18  \
59       60                  Central Park          13948.0       3516.0   
27       28  Gateway - Green Valley Ranch          29201.0      10074.0   
31       32                       Hampden          17547.0       2754.0   
32       33                 Hampden South          14370.0       2639.0   
36       37                       Hilltop           8190.0       2016.0   
12       13                  Cherry Creek           5589.0        413.0   
69       70               Washington Park           6905.0       1199.0   
56       57               South Park Hill           8590.0       1851.0   
74       75                 West Highland           8540.0       1304.0   
35       36                      Highland           8429.0       1296.0   
58       59                         Speer          10954.0        707.0   
19       20                 Congress Park          10235.0       1427.0   
25       26                   Five Points          12712.0       1446.0   
47       48               North Park Hill           9382.0       2179.0   
7         8                      Berkeley           8112.0       1305.0   
41       42                   Lowry Field           8067.0       2000.0   
70       71          Washington Park West           6393.0        807.0   
8         9                  Capitol Hill          14708.0        408.0   
55       56                    Sloan Lake           7238.0       1324.0   
43       44                       Marston          11132.0       2266.0   
68       69              Virginia Village          12844.0       2421.0   
65       66               University Park           7491.0       1135.0   
11       12                 Cheesman Park           7971.0        510.0   
71       72      Washington Virginia Vale          13030.0       2252.0   
26       27                    Fort Logan           8532.0       1796.0   
30       31                          Hale           6936.0        987.0   
46       47            North Capitol Hill           5823.0        194.0   
63       64                    University           9375.0        826.0   
44       45                     Montbello          30348.0      11137.0   
64       65              University Hills           5327.0        925.0   
5         6                   Bear Valley           8889.0       2036.0   
61       62                     Sunnyside           9726.0       2363.0   
77       78                       Windsor          12589.0       1819.0   
34       35             Harvey Park South           8393.0       2065.0   
33       34                   Harvey Park          11525.0       3162.0   
23       24                   East Colfax          10191.0       2595.0   
73       74                   West Colfax           9740.0       2756.0   
0         1                   Athmar Park           8898.0       2649.0   
42       43                       Mar Lee          12452.0       3685.0   
48       49           Northeast Park Hill           7822.0       2198.0   
53       54                     Ruby Hill           9820.0       2865.0   
40       41                  Lincoln Park           6119.0       1365.0   
75       76                      Westwood          15486.0       5678.0   
67       68                    Villa Park           8758.0       2741.0   
29       30                     Goldsmith           5808.0       1060.0   
24       25                Elyria Swansea           6401.0       2322.0   
18       19   College View - South Platte           6498.0       2037.0   
3         4                        Barnum           6111.0       2015.0   

    AGE_18_TO_34  AGE_35_TO_65  AGE_65_PLUS  NUM_HOUSEHOLDS  FAMILIES  \
59        4008.0        6045.0        379.0          4092.0    2759.0   
27        7778.0       10405.0        944.0          9427.0    7056.0   
31        4655.0        6957.0       3181.0          9320.0    4105.0   
32        3729.0        5650.0      

# The chart above displays neighborhoods in Denver with households whose income is above 100k in descending order.

## The top three neighborhoods are **Central Park** with 2,748 households above 100k in income, **Gateway - Green Valley Ranch** with 2,192 households above 100k in income, and **Hampden** with 2,160 households above 100k in income.

# Across both charts the **Gateway - Green Valley Ranch** neighborhood was a close second in target population and having large numbers of households with incomes above 100k.

## Therefore, the most optimal neighborhood for the coffee shop to open stores in should be **Gateway - Green Valley Ranch**.